controller/array_action/utils.py

```diff
@@ -1,11 +1,30 @@
 import encodings
+
+from controller.common.csi_logger import get_stdout_logger
+
 UTF_8 = encodings.utf_8.getregentry().name
 
+logger = get_stdout_logger()
+
 
 def bytes_to_string(input_as_bytes):
     return input_as_bytes.decode(UTF_8) if input_as_bytes else ""
 
 
+def convert_scsi_uuid_to_nguid(vol_id):
+    logger.debug("converting scsi uuid to nguid")
+    indices = [1, 7, 16]
+    splited_text = list()
+    for index, _ in enumerate(indices):
+        last = len(vol_id)
+        if index + 1 < len(indices):
+            last = indices[index + 1]
+        splited_text.append(vol_id[indices[index]:last])
+    final_nguid = splited_text[2] + splited_text[0] + '0' + splited_text[1]

```

Comment:
could we have names for each parsed part, instead of just hardcoded numbers?
also `indices[index]` is confusing.
could you write an example here for a string, the parts to which it needs to be broken down, and how they are composed to the final value?
I'm sure we can find good names for every variable

Reply:
nit: I personally prefer `final_nguid = ''.join(vendor_identifier_extension, oui, '0', vendor_identifier`)

Reply:
we can also consider using consts, e.g. `WWN_OUI_END = 7`

Reply:
done